sunpy/instr/goes.py

```diff
@@ -88,6 +89,7 @@
 GOES_REMOTE_PATH = "http://{0}/ssw/gen/idl/synoptic/goes/".format(HOST)
 # Define location where data files should be downloaded to.
 DATA_PATH = config.get("downloads", "download_dir")
+create_download_dir()
```

Comment:
There is no need for two function calls here (get default directory and create it) this can be done in one call.

Also, this is still to early to create the directory. We only want to be creating it upon use, so if the functions are called with the default data dir we should make the directory then, rather than on import of this submodule.

To give you an example for this file you want to replace every occurrence of `DATA_PATH` with `sunpy.config.get_and_create_download_dir()`.